libraries/aws_iam_access_keys.rb

```diff
@@ -86,16 +88,23 @@ class AccessKeyProvider
     class AwsUserIterator < AccessKeyProvider
       def fetch(criteria)
         iam_client = AWSConnection.new.iam_client
-        usernames = []
+
+        user_details = {}
         if criteria.key?(:username)
-          usernames.push criteria[:username]
+          begin
+            user_details[criteria[:username]] = iam_client.get_user(user_name: criteria[:username]).user
+          rescue Aws::IAM::Errors::NoSuchEntity # rubocop:disable Lint/HandleExceptions
+            # Swallow - a miss on search results should return an empty table
```

Comment:
I'm not sure about masking the error here, but don't have any other suggestions. I'd get another opinion on it.

Reply:
Well, this is a plural resource.  A common use case is enforcing the non-existence of something.  So, we might send a request to AWS that results in no results.  The behavior varies from service to service and method call to method call, but in this case, the SSDK throws an exception.  But for the control/profile author, not finding anything isn't an error, it just means that `exists` should return false.  So, that's what we do.

In this particular case - you wanted to know if a particular user has any keys, and it turns our that particular user doesn't even exist - well, there you have it: no such keys exist, either.